packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/CallbackExtensions.cs

```diff
@@ -65,37 +67,95 @@ static CallbackResult InvokeCallbackCore(this Callback callback, IReferenceMap r
             throw new ArgumentException("Callback does not specificy a method, getter, or setter to invoke");
         }
 
-        static CallbackResult InvokeMethod(InvokeRequest request, IReferenceMap referenceMap)
+        private static CallbackResult InvokeMethod(InvokeRequest request, IReferenceMap referenceMap)
         {
             request = request ?? throw new ArgumentNullException(nameof(request));
-            DeputyBase deputy = referenceMap.GetOrCreateNativeReference(request.ObjectReference);
+            var deputy = referenceMap.GetOrCreateNativeReference(request.ObjectReference);
 
-            MethodInfo methodInfo = ReflectionUtils.GetNativeMethod(deputy.GetType(), request.Method);
+            var methodInfo = ReflectionUtils.GetNativeMethod(deputy.GetType(), request.Method);
 
             if (methodInfo == null)
             {
                 throw new InvalidOperationException($"Received callback for {deputy.GetType().Name}.{request.Method} method, but this method does not exist");
             }
 
-            JsiiMethodAttribute attribute = methodInfo.GetCustomAttribute<JsiiMethodAttribute>();
-            var returnValue = methodInfo.Invoke(deputy, request.Arguments.Select(arg => FromKernel(arg, referenceMap)).ToArray());
-            return attribute?.Returns != null ? new CallbackResult(attribute?.Returns, returnValue) : null;
+            var attribute = methodInfo.GetAttribute<JsiiMethodAttribute>();
+            var parameters = methodInfo.GetParameters();
+            
+            var converter = ServiceContainer.ServiceProvider.GetRequiredService<IJsiiToFrameworkConverter>();
+            
+            var rehydratedArgs = Enumerable.Range(0, request.Arguments.Length)
```

Comment:
All this work should be in its own function, no? I have a hard time grokking it and I trust you that it's correct, but I'd like to see it factored out into more readable code units.

Seems like the same hydration/dehydration should happen in multiple places, so why isn't the code shared?